imageloading-core/src/main/java/com/google/accompanist/imageloading/Image.kt

```diff
@@ -31,162 +31,50 @@ import androidx.compose.runtime.rememberCoroutineScope
-            // Re-throw all IllegalStateExceptions
-            throw e
-        } catch (e: IllegalArgumentException) {
-            // Re-throw all IllegalArgumentExceptions
-            throw e
-        } catch (t: Throwable) {
-            // Anything else, we wrap in a Error state instance
-            ImageLoadState.Error(result = null, throwable = t, request = request)
-        }
-    }
-
-    /**
-     * Extending classes should implement this function to execute the [request] with the given
-     * [size] constraints.
-     */
-    protected abstract suspend fun executeRequest(request: R, size: IntSize): ImageLoadState
-
-    internal val internalRequestFlow: Flow<R?> get() = snapshotFlow { request }
-    internal val internalRequest get() = request
-}
-
-/**
- * A generic image loading composable, which provides hooks for image loading libraries to use.
+ * A generic image loading painter, which provides hooks for image loading libraries to use.
  * Apps shouldn't generally use this function, instead preferring one of the extension libraries
  * which build upon this, such as the Coil and Glide libraries.
  *
  * @param state The request to execute.
- * @param contentDescription text used by accessibility services to describe what this image
- * represents. This should always be provided unless this image is used for decorative purposes,
- * and does not represent a meaningful action that a user can take. This text should be
- * localized, such as by using [androidx.compose.ui.res.stringResource] or similar.
- * @param modifier [Modifier] used to adjust the layout algorithm or draw decoration content.
- * @param alignment Optional alignment parameter used to place the loaded [ImageBitmap] in the
- * given bounds defined by the width and height.
- * @param contentScale Optional scale parameter used to determine the aspect ratio scaling to be
- * used if the bounds are a different size from the intrinsic size of the loaded [ImageBitmap].
- * @param colorFilter Optional [ColorFilter] to apply for the [Painter] when it is rendered onscreen.
- * If you provide a color filter and enable [fadeIn], the content will be drawn in a off-screen
- * layer whilst the animation is running, which is known is cause performance issues.
  * @param fadeIn Whether to run a fade-in animation when images are successfully loaded.
  * Default: `false`.
  * @param fadeInDurationMs Duration for the fade animation in milliseconds when [fadeIn] is enabled.
  * @param previewPlaceholder Drawable resource ID which will be displayed when this function is
  * ran in preview mode.
  */
 @Composable
-fun <R : Any> Image(
+fun <R : Any> rememberLoadPainter(
     state: ImageState<R>,
-    contentDescription: String?,
-    modifier: Modifier = Modifier,
-    alignment: Alignment = Alignment.Center,
-    contentScale: ContentScale = ContentScale.Fit,
-    colorFilter: ColorFilter? = null,
     fadeIn: Boolean = false,
     fadeInDurationMs: Int = DefaultTransitionDuration,
```

Comment:
it is usually preferred to not use the internal vals as a default values as users can't really use it. imagine someone wants to write something like this
rememberLoadPainter(fadeInDurationMs = if (haveCustomDuration) customDuration else DefaultTransitionDuration)
and they can't do so. we usually or make a public constant or just inline the value. 
I am also curious is 1000 really a good default? isnt it too long?

Reply:
Extracted this default value to a `Defaults` object